packages/webdriverio/tests/commands/browser/url.test.js

```diff
@@ -27,14 +27,39 @@ describe('url', () => {
             .toEqual({ url: 'http://foobar.com/foobar' })
     })
 
-    it('should throw an exception when a non-string value passed in', async () => {
+    it('should throw an exception when a non-string value passed in path', async () => {
         try {
             await browser.url(true)
         } catch (e) {
             expect(e.message).toContain('command needs to be type of string')
         }
     })
 
+    it('should throw an exception when a non-function value passed in inject', async () => {
+        try {
+            await browser.url('/', { inject: 123 })
+        } catch (e) {
+            expect(e.message).toContain('command needs to be type of function')
+        }
+    })
+
+    it('should continue the navigation if inject parameter is not passed', async () => {
+        await browser.url('/')
+        expect(got.mock.calls[0][1].json).toEqual({ url: 'http://foobar.com/' })
+    })
+
+    it('should accept an arrow function as the inject parameter', async () => {
+        const arrowFunction = jest.fn(() => console.log('foo-bar'))
+        await browser.url('/', arrowFunction)
+        expect(got.mock.calls[0][1].json).toEqual({ url: 'http://foobar.com/' })
```

Comment:
let's also add a test that we used puppeteer to inject the method

Reply:
Same here - What would be the way to check this. Sorry, new to unit tests.

Reply:
mock puppeteer similar as I did above and then check if evaluate of page has been called with that property:

```js
expect(pageMock).toBeCalledWith(arrowFunction)
```